sklearn/calibration.py

```diff
@@ -202,11 +188,47 @@ def fit(self, X, y, sample_weight=None):
             base_estimator = self.base_estimator
 
         if self.cv == "prefit":
+            # Set `n_features_in_` attribute
+            try:
+                check_is_fitted(self)
+            except NotFittedError as nfe:
+                raise AttributeError(
+                    "{} object has no n_features_in_ attribute."
+                    .format(self.__class__.__name__)
+                ) from nfe
```

Comment:
I _think_ that this error should be tested.
You might have a look to test coverage for your file [here](https://codecov.io/gh/scikit-learn/scikit-learn/compare/c3835442e50077d32b1b5f9771e7c6b7b999bbfc...e3499ea855599c01cc261077f63d574d4102b152/src/sklearn/calibration.py?before=sklearn/calibration.py). When it's red it means that a test should be added.

Reply:
@glemaitre why are we raising an `AttributeError` here? Shouldn't we be raising the `NotFittedError` as this is the cause of `n_features_in_` not appearing..?

Reply:
It is to simulate that the `n_features_in_` does not exist which would raise an `AttributeError` in a normal estimator.